src/frontend/src/handler/create_source.rs

```diff
@@ -99,11 +99,7 @@ async fn extract_avro_table_schema(
     info: &StreamSourceInfo,
     with_properties: &HashMap<String, String>,
 ) -> Result<Vec<ColumnCatalog>> {
-    let parser_config = SpecificParserConfig::new(
-        SourceStruct::new(SourceFormat::Plain, SourceEncode::Avro),
-        info,
-        with_properties,
-    )?;
+    let parser_config = SpecificParserConfig::new(info, with_properties)?;
```

Comment:
One of the caller is actually passing in UpsertAvro today. Is the caller supposed to call `extract_upsert_avro_table_schema`?

It does not affect correctness right now because only `encoding_properties` field is used, and `SpecificParserConfig::new` handles UpsertAvro and PlainAvro the same for encoding.

https://github.com/risingwavelabs/risingwave/blob/6fb7ae5f47addc3953e3be76778c70dd611a5bf4/src/frontend/src/handler/create_source.rs#L532-L559

Reply:
Dates back to first day upsert was supported: #8111